sdk/search/azure-search-documents/azure/search/documents/indexes/_internal/_models.py

```diff
@@ -4,7 +4,142 @@
+     "en.microsoft", "en.lucene", "et.microsoft", "fi.microsoft", "fi.lucene", "fr.microsoft",
+     "fr.lucene", "gl.lucene", "de.microsoft", "de.lucene", "el.microsoft", "el.lucene",
+     "gu.microsoft", "he.microsoft", "hi.microsoft", "hi.lucene", "hu.microsoft", "hu.lucene",
+     "is.microsoft", "id.microsoft", "id.lucene", "ga.lucene", "it.microsoft", "it.lucene",
+     "ja.microsoft", "ja.lucene", "kn.microsoft", "ko.microsoft", "ko.lucene", "lv.microsoft",
+     "lv.lucene", "lt.microsoft", "ml.microsoft", "ms.microsoft", "mr.microsoft", "nb.microsoft",
+     "no.lucene", "fa.lucene", "pl.microsoft", "pl.lucene", "pt-BR.microsoft", "pt-BR.lucene", "pt-
+     PT.microsoft", "pt-PT.lucene", "pa.microsoft", "ro.microsoft", "ro.lucene", "ru.microsoft",
+     "ru.lucene", "sr-cyrillic.microsoft", "sr-latin.microsoft", "sk.microsoft", "sl.microsoft",
+     "es.microsoft", "es.lucene", "sv.microsoft", "sv.lucene", "ta.microsoft", "te.microsoft",
+     "th.microsoft", "th.lucene", "tr.microsoft", "tr.lucene", "uk.microsoft", "ur.microsoft",
+     "vi.microsoft", "standard.lucene", "standardasciifolding.lucene", "keyword", "pattern",
+     "simple", "stop", "whitespace".
+    :type analyzer_name: str or ~azure.search.documents.indexes.models.LexicalAnalyzerName
+    :param tokenizer_name: The name of the tokenizer to use to break the given text. If this parameter
+     is not specified, you must specify an analyzer instead. The tokenizer and analyzer parameters
+     are mutually exclusive. Possible values include: "classic", "edgeNGram", "keyword_v2",
+     "letter", "lowercase", "microsoft_language_tokenizer", "microsoft_language_stemming_tokenizer",
+     "nGram", "path_hierarchy_v2", "pattern", "standard_v2", "uax_url_email", "whitespace".
+    :type tokenizer_name: str or ~azure.search.documents.indexes.models.LexicalTokenizerName
+    :param token_filters: An optional list of token filters to use when breaking the given text.
+     This parameter can only be set when using the tokenizer parameter.
+    :type token_filters: list[str or ~azure.search.documents.indexes.models.TokenFilterName]
+    :param char_filters: An optional list of character filters to use when breaking the given text.
+     This parameter can only be set when using the tokenizer parameter.
+    :type char_filters: list[str]
+    """
+
+    _validation = {
+        'text': {'required': True},
+    }
+
+    _attribute_map = {
+        'text': {'key': 'text', 'type': 'str'},
+        'analyzer_name': {'key': 'analyzerName', 'type': 'str'},
+        'tokenizer_name': {'key': 'tokenizerName', 'type': 'str'},
+        'token_filters': {'key': 'tokenFilters', 'type': '[str]'},
+        'char_filters': {'key': 'charFilters', 'type': '[str]'},
+    }
+
+    def __init__(
+        self,
+        **kwargs
+    ):
+        super(AnalyzeTextOptions, self).__init__(**kwargs)
+        self.text = kwargs['text']
+        self.analyzer_name = kwargs.get('analyzer_name', None)
```

Comment:
Just for my own understanding - why `get` here vs `pop` in search when taking apart kwargs?

Reply:
The difference is pop will remove the argument from kwargs. In init method, kwargs will not be used in anywhere else so there is no difference. But in e.g. search method, if I don't pop them, those arguments will be passed into transport which will cause problems.